src/community/label_propagation.jl

```diff
@@ -0,0 +1,81 @@
+"""
+Community detection using the label propagation algorithm (see [Raghavan et al.](http://arxiv.org/abs/0709.2938)).
+`g`: imput Graph
+`ϵ`: proportion of unsatisfied nodes
+return : array containing vertex assignments
+"""
+function label_propagation(g::SimpleGraph, ϵ=1.0e-8)
+    n = nv(g)
+    label = collect(1:n)
+    runing_nodes = Set(vertices(g))
+
+    while length(runing_nodes) > n*ϵ
+        order = shuffle(collect(runing_nodes))
+        for u in order
+            if update_label!(g, label, u)
+                for v in out_neighbors(g, u)
+                    push!(runing_nodes, v)
+                end
+            else
+                delete!(runing_nodes, u)
+            end
+        end
+    end
+    permute_labels!(label)
+    label
+end
+
+function label_count(g::SimpleGraph, membership::Vector{Int}, u::Int)
+    label_cnt = Dict{Int, Int}()
+    for v in out_neighbors(g,u)
+        v_lbl = membership[v]
+        label_cnt[v_lbl] = get(label_cnt, v_lbl, 0) + 1
+    end
+    label_cnt
+end
+
+function update_label!(g::SimpleGraph, membership::Vector{Int}, u::Int)
+	running = false
+	if degree(g, u) > 0
+	    old_lbl = membership[u]
+	    label_cnt = label_count(g, membership, u)
+	    neigh_lbls = collect(keys(label_cnt))
+	    neigh_lbls_cnt = collect(values(label_cnt))
```

Comment:
I am concerned about the number of allocations here. It looks like 3 allocations for each vertex for each time around the loop over vertices. Perhaps using one big `labelcounts = Dict{(Int, Int), Int}` is a better option. So you would have `labelcounts[(u, lab)] = labelcnt[lab]` this would be allocated once per loop.

But I guess we can put that in as an optimization once someone needs it.